### PR TITLE
Fix category contiguity checking logic in `_should_use_raw_category_ids`

### LINKED ISSUE #262: Should label ids start from 0?
There are many issues about class number reinitialization. I would like to point out that this repo works as expected when labels start from 0. In terminal i get Namespace(num_classes=2,...) which is the correct value. I have 2 classes. 

When labels start from 1 i get a wrong value Namespace(num_classes=3,...) . I am not sure if this will cause problems as i was previously always worked with labels starting from 0. I also made a pull request, where i corrected warnings, but maybe the problem is deeper.

### PR DESCRIPTION
**Pull request by Abdul-Mukit**
## What does this PR do?
Fixes: https://github.com/roboflow/rf-detr/issues/262

Initial bug reported in https://github.com/roboflow/rf-detr/issues/262#issuecomment-3945932265
### Updated problem description:  
When training with a 0-indexed roboflow-formatted dataset, both loss and evaluation metrics progress normally.  
When training with a 1-indexed roboflow-formatted dataset, only the losses go down normally; evaluation metrics don't go up as expected. 

### Root cause:
Losses go down normally in both cases, as we force `CocoDetection(remap_category_ids=True)` in  `build_roboflow_from_coco`.  
This causes `targets`->`labels` from dataloaders to always be remapped and start from 0. In other words, the 0th classification neuron is trained for the same object category in both dataset cases.  

However, category_val.py -> `_should_use_raw_category_ids`, has a bug.
```
# If any label is a COCO category ID but not a valid contiguous index,
# treat the whole run as raw-ID output to avoid corrupting eval categories.
uses_raw_ids = any((label in self.cat_ids) and (label not in self.label2cat) for label in labels)
if uses_raw_ids:
        self._prefer_raw_category_ids = True
        return True
```
Example:  
We have 4 categories.  
cat_ids = {1, 2, 3, 4}
self.label2cat = {0: 1, 1: 2, 2: 3, 3: 4}.  
On the first call to `_should_use_raw_category_ids()`, we got at least one 4 in the `labels`. Resulting in `uses_raw_ids = True`. However, in the `CocoDetection`, we already made sure remapping is enforced for Roboflow datasets `CocoDetection(remap_category_ids=True)`. Getting a noisy label 4 at the first call to `_should_use_raw_category_ids` causes a faulty guess. Actually, `uses_raw_ids` should be False for the above case, as we see a difference.

If `cat_ids = {0, 1, 2, 3}` and `self.label2cat = {0: 0, 1: 1, 2: 2, 3: 3}`, the current logic will return `uses_raw_ids=False`. This should be True as using this `label2cat` will result in no change.  

### Proposed solution:
The solution is to check whether keys and values of `label2cat` (if it was created by `CocoDetection(remap_category_ids=True)`), have any difference. If not, we can use raw-id.  

 

**Related Issue(s):** 

## Type of Change

- Bug fix (non-breaking change that fixes an issue)

## Testing

- [x] I have tested this change locally
- [x] I have added/updated tests for this change

**Test details:**

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code where necessary, particularly in hard-to-understand areas
- [x] My changes generate no new warnings or errors
- [ ] I have updated the documentation accordingly (if applicable)

### CODE REVIEW

**Comment by Abdul-Mukit** [comment]
I'll work on the tests.

File: src/rfdetr/datasets/coco_eval.py [hunk 1]
```diff
@@ -89,9 +89,9 @@ def _should_use_raw_category_ids(self, labels: List[int]) -> bool:
         if self._prefer_raw_category_ids:
             return True
 
-        # If any label is a COCO category ID but not a valid contiguous index,
-        # treat the whole run as raw-ID output to avoid corrupting eval categories.
-        uses_raw_ids = any((label in self.cat_ids) and (label not in self.label2cat) for label in labels)
+        # The labels and categories differ when categories are not contiguous starting from 0.
+        # If labels and categories are same in label2cat, we can simply use raw-ID.
+        uses_raw_ids = list(self.label2cat.keys()) == list(self.label2cat.values())
```
[the remark this answers is not in the document]
Comment (Abdul-Mukit) [reply src/rfdetr/datasets/coco_eval.py:99]: Done.

**Review by Borda — APPROVED** [review]

**Comment by Abdul-Mukit** [comment]
Thank you. I learned new skills. Looking forward to more.
